code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/compatibility/CacheManager.java

```diff
@@ -0,0 +1,417 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.internal.compatibility;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.adobe.marketing.mobile.LoggingMode;
+import com.adobe.marketing.mobile.MobileCore;
+import com.adobe.marketing.mobile.services.CacheFileService;
+import com.adobe.marketing.mobile.internal.utility.FileUtils;
+import com.adobe.marketing.mobile.internal.utility.StringEncoder;
+import com.adobe.marketing.mobile.internal.utility.StringUtils;
+import com.adobe.marketing.mobile.services.DeviceInforming;
+
+import java.io.File;
+import java.util.Map;
+
+/**
+ * Legacy cache file management class. Only exists to allow compatibility with
+ * Configuration and Campaign extension for using {@link com.adobe.marketing.mobile.utils.RemoteDownloader}.
+ * NOTE: Bulk of the logic in this class has been copied from
+ * {@link com.adobe.marketing.mobile.CacheManager}.
+ */
+public class CacheManager implements CacheFileService {
```

Comment:
@praveek should we move this compatibility class to the legacy directory: https://github.com/adobe/aepsdk-core-android/tree/feature/eventhub/code/android-core-library/src/legacy?